src/herder/HerderSCPDriver.cpp

```diff
@@ -687,59 +691,34 @@ HerderSCPDriver::combineCandidates(uint64_t slotIndex,
     }
 
     // take the txSet with the biggest size, highest xored hash that we have
-    TxSetFramePtr bestTxSet;
+    using HighestPair = std::pair<StellarValue, TxSetFramePtr>;
+    HighestPair highestPair = std::accumulate(
```

Comment:
using the std library seems fine here... but it looks like it's not a home run:
* `std::accumulate` doesn't seem to be a great fit though as it copies the value all the time (which contains a hash). There is a version that uses `std::move` but that's C++ 20.
* there is `std::reduce` but that's C++17
* from what I see the only thing we can do is a straight loop at this time 🤷

Reply:
Alright, I gave up on the STL.  I kept everything except the two accumulators (the pointers to the highest value and set) `const`, though (including the iterator, i.e. `cbegin()` and `cend()`).